README.md

```diff
@@ -112,6 +113,16 @@ s = unstructured_client.UnstructuredClient(client: http_client)
 
 <!-- Placeholder for Future Speakeasy SDK Sections -->
 
+### Parameters
+
+`PartitionParameters` has parameters listed in `docs/models/shared/partitionparameters.md`
+
+More details about:
+1. `split_pdf_page` - when you set to True, it will cause the pdf to be split at client side, 
+   before sending to api. This currently causes the `parent_id` generation to be disabled, as that
+   requires having context of multiple pages. **Also this will fail for non-pdf files, so set it to
+   True only for pdfs.** The amount of threads that will be used for sending individual pdf pages, is controlled by `UNSTRUCTURED_CLIENT_SPLIT_CALL_THREADS` env var. By default it equals to 5.
```

Comment:
I recognize the bolded part is major downside of this code. It's because it's difficult to recognize if file is pdf, as AFAIK it's not done currently at client. And I don't want to check extension, as it can be missing, plus a lot of other special cases which are handled by backend. Unless it's agreed to be better solution to at least try with the extension, because it probably is assigned in 95% of cases. Other option is to check for PdfStreamError as in test_decorators.py , and if it raises, then steer the partitioning to non-split version. WDYT?

Reply:
I think there can be cases where it would make sense to duplicate the code instead of importing a whole package, but I'm fine either way. Was just an idea for more safely handling incorrect files passed unintentionally.

Reply:
Yes it's important problem here, as I'm sure we'll get lots of irritated users pointing this out. Sure, I'm also for copying small function or two, however, looking at filetype.py it's big, it's not everything-in-single file, it does have a lot of imports, and it will change often future, so unless it's extracted into separate util library, it's too heavy.  

So IMHO the options are two (and could be combined) 1) looking for PdfStreamError and then disabling splitting 2) checking extension- if it's not .pdf then disable splitting. Should cover 98% of cases. @awalker4 ?

Reply:
Ok, implemented it.